GEOSwift/MapKit.swift

```diff
@@ -8,100 +8,35 @@
-// Solve conformance to protocol with an inelegant (but effective) switch case.
-
-extension Geometry: GEOSwiftMapKit {
+extension Geometry {
     public func mapShape() -> MKShape {
-
         switch self {
-
         case is Waypoint:
             let pointAnno = MKPointAnnotation()
-            pointAnno.coordinate = CLLocationCoordinate2DFromCoordinate((self as! Waypoint).coordinate)
+            pointAnno.coordinate = CLLocationCoordinate2D((self as! Waypoint).coordinate)
             return pointAnno
-
         case is LineString:
-            var coordinates = (self as! LineString).points.map({ (point: Coordinate) ->
-                CLLocationCoordinate2D in
-                CLLocationCoordinate2DFromCoordinate(point)
-            })
-            let polyline = MKPolyline(coordinates: &coordinates,
-                                      count: coordinates.count)
-            return polyline
-
+            var coordinates = (self as! LineString).points.map(CLLocationCoordinate2D.init)
+            return MKPolyline(coordinates: &coordinates,
+                              count: coordinates.count)
         case is Polygon:
-            var exteriorRingCoordinates = (self as! Polygon).exteriorRing.points.map({ (point: Coordinate) ->
-                CLLocationCoordinate2D in
-                CLLocationCoordinate2DFromCoordinate(point)
-            })
-
-            let interiorRings = (self as! Polygon).interiorRings.map({ (linearRing: LinearRing) ->
-                MKPolygon in
-                MKPolygonWithCoordinatesSequence(linearRing.points)
-            })
-
-            let polygon = MKPolygon(coordinates: &exteriorRingCoordinates,
-                                    count: exteriorRingCoordinates.count,
-                                    interiorPolygons: interiorRings)
-            return polygon
-
-        case let gc as GeometryCollection<Waypoint>:
+            var exteriorRingCoordinates = (self as! Polygon).exteriorRing.points.map(CLLocationCoordinate2D.init)
+            let interiorRings = (self as! Polygon).interiorRings.map {
+                MKPolygonWithCoordinatesSequence($0.points)
+            }
+            return MKPolygon(coordinates: &exteriorRingCoordinates,
+                             count: exteriorRingCoordinates.count,
+                             interiorPolygons: interiorRings)
+        case let gc as GeometryCollection<Geometry>:
             return MKShapesCollection(geometryCollection: gc)
-
-        case let gc as GeometryCollection<LineString>:
-            return MKShapesCollection(geometryCollection: gc)
-
-        case let gc as GeometryCollection<Polygon>:
-            return MKShapesCollection(geometryCollection: gc)
-
         default:
-            let geometryCollectionOverlay = MKShapesCollection(geometryCollection: (self as! GeometryCollection))
-            return geometryCollectionOverlay
+            fatalError("unhandled case")
```

Comment:
There shouldn't be a way to actually hit this case unless someone adds their own Geometry subclass. I'm not sure what we'd return in that case though. Maybe we could allow this method to return nil?

Reply:
Is there an empty shape? We could consider returning that so it doesn't crash and we don't have to deal with making it optional.

Reply:
I don't know the answer. @vfn?

Reply:
This method actually returns MKShapes (I was thinking Geometry). I suppose we could create an MKShape subclass that does nothing…

Reply:
I think it'd actually be easier to just make it return an optional. This changes the API, but we already have a number of other breaking changes in develop, and I think it's a more accurate description of the behavior.

Reply:
Okay, I pushed a commit to allow this method to return nil